tests/avocado/avocado_qemu/test.py

```diff
@@ -342,6 +341,24 @@ def migrate_complete():
 
         return newvm
 
+    def add_image(self, path, username=None, password=None, snapshot=True,
+                  extra=None):
+        file_option = 'file=%s' % path
+        for item in self.args:
+            if file_option in item:
+                logging.error('Image %s already present', path)
+                return
+
+        if extra is not None:
+            file_option += ',%s' % extra
+        self.args.extend(['-drive', file_option])
+
+        if snapshot:
+            self.args.append('-snapshot')
+
+        self.username = username
```

Comment:
I'd only update the `username/password` when it's not `None`. Some people might want to add multiple images and specify the user/pass only once ...